### ISSUE TITLE
Unknown element type specified!

### ISSUE THREAD

**Issue by the-nic** [fusiondirectory/fusiondirectory#34]
Using php 7.0.13 on ubuntu 16.04 I get the following error when I try to open any user or groups tab (fusiondirectory 1.0.19 from jessie packages):

`Unknown element type specified!`
The stack trace is 
```
#0  filter->render() called at [/usr/share/fusiondirectory/include/class_listing.inc:507]
#1  listing->render() called at [/usr/share/fusiondirectory/include/class_management.inc:216]
#2  management->renderList() called at [/usr/share/fusiondirectory/include/simpleplugin/class_simpleManagement.inc:411]
#3  simpleManagement->renderList() called at [/usr/share/fusiondirectory/plugins/admin/users/class_userManagement.inc:119]
#4  userManagement->renderList() called at [/usr/share/fusiondirectory/include/class_management.inc:210]
#5  management->execute() called at [/usr/share/fusiondirectory/plugins/admin/users/main.inc:47]
#6  require(/usr/share/fusiondirectory/plugins/admin/users/main.inc) called at [/usr/share/fusiondirectory/html/main.php:286]
```

And dumping `$this->elements` gives

```
array(19) {
  ["FILTERACLASSIGNMENT"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(19) "FILTERACLASSIGNMENT"
    ["default"]=>
    bool(true)
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(21) "(objectClass=gosaAcl)"
  }
  ["FILTERORGANIZATION"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(18) "FILTERORGANIZATION"
    ["default"]=>
    bool(true)
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(57) "(&(objectClass=organization)(objectClass=gosaDepartment))"
  }
  ["FILTERDEPARTMENT"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(16) "FILTERDEPARTMENT"
    ["default"]=>
    bool(true)
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(63) "(&(objectClass=organizationalUnit)(objectClass=gosaDepartment))"
  }
  ["FILTERLOCALITY"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(14) "FILTERLOCALITY"
    ["default"]=>
    bool(true)
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(53) "(&(objectClass=locality)(objectClass=gosaDepartment))"
  }
  ["FILTERDCOBJECT"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(14) "FILTERDCOBJECT"
    ["default"]=>
    bool(true)
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(53) "(&(objectClass=dcObject)(objectClass=gosaDepartment))"
  }
  ["FILTERCOUNTRY"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(13) "FILTERCOUNTRY"
    ["default"]=>
    bool(true)
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(52) "(&(objectClass=country)(objectClass=gosaDepartment))"
  }
  ["FILTERDOMAIN"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(12) "FILTERDOMAIN"
    ["default"]=>
    bool(true)
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(51) "(&(objectClass=domain)(objectClass=gosaDepartment))"
  }
  ["WORKSTATION"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(11) "WORKSTATION"
    ["default"]=>
    string(4) "true"
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(29) "(objectClass=gotoWorkstation)"
  }
  ["MOBILEPHONE"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(11) "MOBILEPHONE"
    ["default"]=>
    string(4) "true"
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(27) "(objectClass=fdMobilePhone)"
  }
  ["WINSTATION"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(10) "WINSTATION"
    ["default"]=>
    string(4) "true"
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(41) "(&(objectClass=sambaSamAccount)(uid=*\$))"
  }
  ["COMPONENT"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(9) "COMPONENT"
    ["default"]=>
    string(4) "true"
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(50) "(&(objectClass=ieee802Device)(objectClass=device))"
  }
  ["TERMINAL"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(8) "TERMINAL"
    ["default"]=>
    string(4) "true"
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(26) "(objectClass=gotoTerminal)"
  }
  ["PRINTER"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(7) "PRINTER"
    ["default"]=>
    string(4) "true"
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(23) "(objectClass=fdPrinter)"
  }
  ["SERVER"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(6) "SERVER"
    ["default"]=>
    string(4) "true"
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(22) "(objectClass=goServer)"
  }
  ["PHONE"]=>
  array(5) {
    ["type"]=>
    string(8) "checkbox"
    ["tag"]=>
    string(5) "PHONE"
    ["default"]=>
    string(4) "true"
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(21) "(objectClass=fdPhone)"
  }
  ["NAME"]=>
  array(9) {
    ["type"]=>
    string(9) "textfield"
    ["tag"]=>
    string(4) "NAME"
    ["size"]=>
    string(2) "20"
    ["maxlength"]=>
    string(2) "60"
    ["default"]=>
    array(0) {
    }
    ["unset"]=>
    array(0) {
    }
    ["set"]=>
    string(58) "(|(dn=*$*)(cn=*$*)(description=*$*)(ou=*$*)(dc=*$*)(=*$*))"
    ["alphabet"]=>
    string(5) "false"
    ["autocomplete"]=>
    array(5) {
      ["backend"]=>
      string(7) "SYSTEMS"
      ["filter"]=>
      string(108) "(&(|$FILTERACLASSIGNMENT)(|(dn=*$NAME*)(cn=*$NAME*)(description=*$NAME*)(ou=*$NAME*)(dc=*$NAME*)(=*$NAME*)))"
      ["attribute"]=>
      array(6) {
        [0]=>
        string(2) "dn"
        [1]=>
        string(2) "cn"
        [2]=>
        string(11) "description"
        [3]=>
        string(2) "ou"
        [4]=>
        string(2) "dc"
        [5]=>
        bool(false)
      }
      ["frequency"]=>
      string(3) "0.5"
      ["characters"]=>
      string(1) "3"
    }
  }
  ["FUNCTIONAL"]=>
  array(1) {
    ["set"]=>
    string(61) "(!(|(objectClass=posixAccount)(objectClass=gosaMailAccount)))"
  }
  ["MAIL"]=>
  array(2) {
    ["unset"]=>
    string(0) ""
    ["set"]=>
    string(0) ""
  }
  ["SAMBA"]=>
  array(2) {
    ["unset"]=>
    string(0) ""
    ["set"]=>
    string(0) ""
  }
}
```

Apperently, neither `MAIL` nor `SAMBA` have the `type` attribute leading to this error. Any idea?

**Comment by the-nic** [comment]
Really weird. After some time this issue simply vanished.

**Comment by coudot** [comment]
Note that I also face this issue but I can't manage to have a set-by-step reproduction.

**Comment by bilbo-the-hobbit** [comment]
hello,

yes we found a bug report un debian also https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=847571

but its seems the uploaded package doesnt fix the issue 

Cheers

**Comment by MCMic** [comment]
It’s weird, looks like the elements for systems management class end up in the filter of users management class.
I’m not sure how this is possible. Filter is stored in the session but it’s supposed to be stored under its management class name so there should be no mix-ups like this one.

**Comment by the-nic** [comment]
This happened after removing a user. And now it occurs again after creating a group.

**Comment by MCMic** [comment]
It seems several people are experiencing this, did someone find a way to reliably reproduce this?

**Comment by DeMiNe0** [comment]
I'm experiencing this issue, but don't have a reliable step based way to reproduce it. I can generally always get to this error by creating, deleting or modifying users or possibly groups/departments. I've also seen an error pop up regarding user filters too, 

```
root@DIRECTORY-01:~# php -v
PHP 7.1.1-1+deb.sury.org~xenial+1 (cli) (built: Jan 20 2017 09:20:20) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.1.1-1+deb.sury.org~xenial+1, Copyright (c) 1999-2017, by Zend Technologies
root@DIRECTORY-01:~# cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.1 LTS"
root@DIRECTORY-01:~# dpkg -l | grep php
ii  libapache2-mod-php                 1:7.0+35ubuntu6                     all          server-side, HTML-embedded scripting language (Apache 2 module) (default)
ii  libapache2-mod-php7.0              7.0.13-0ubuntu0.16.04.1             amd64        server-side, HTML-embedded scripting language (Apache 2 module)
ii  libapache2-mod-php7.1              7.1.1-1+deb.sury.org~xenial+1       amd64        server-side, HTML-embedded scripting language (Apache 2 module)
ii  php                                1:7.1+49+deb.sury.org~xenial+4      all          server-side, HTML-embedded scripting language (default)
ii  php-common                         1:35ubuntu6                         all          Common files for PHP packages
ii  php-curl                           1:7.1+49+deb.sury.org~xenial+4      all          CURL module for PHP [default]
ii  php-fpdf                           3:1.7.dfsg-1.1ubuntu1               all          PHP class to generate PDF files
ii  php-gd                             1:7.1+49+deb.sury.org~xenial+4      all          GD module for PHP [default]
ii  php-imagick                        3.4.3~rc2-2+deb.sury.org~xenial+1   amd64        Provides a wrapper to the ImageMagick library
ii  php-imap                           1:7.1+49+deb.sury.org~xenial+4      all          IMAP module for PHP [default]
ii  php-intl                           1:7.0+35ubuntu6                     all          Internationalisation module for PHP [default]
ii  php-ldap                           1:7.0+35ubuntu6                     all          LDAP module for PHP [default]
ii  php-mbstring                       1:7.0+35ubuntu6                     all          MBSTRING module for PHP [default]
ii  php-mcrypt                         1:7.0+35ubuntu6                     all          libmcrypt module for PHP [default]
ii  php-mdb2                           2.5.0b5-1build1                     all          merge of the PEAR DB and Metabase php database abstraction layers
ii  php-pear                           1:1.10.1+submodules+notgz-6         all          PEAR Base System
ii  php-recode                         1:7.1+49+deb.sury.org~xenial+4      all          recode module for PHP [default]
ii  php-xml                            1:7.0+35ubuntu6                     all          DOM, SimpleXML, WDDX, XML, and XSL module for PHP [default]
ii  php5-common                        5.5.9+dfsg-1ubuntu4.20              amd64        Common files for packages built from the php5 source
ii  php5-curl                          5.5.9+dfsg-1ubuntu4.20              amd64        CURL module for php5
ii  php5-json                          1.3.2-2build1                       amd64        JSON module for php5
ii  php5-mysql                         5.5.9+dfsg-1ubuntu4.20              amd64        MySQL module for php5
ii  php7.0-cli                         7.0.13-0ubuntu0.16.04.1             amd64        command-line interpreter for the PHP scripting language
ii  php7.0-common                      7.0.13-0ubuntu0.16.04.1             amd64        documentation, examples and common module for PHP
ii  php7.0-intl                        7.0.13-0ubuntu0.16.04.1             amd64        Internationalisation module for PHP
ii  php7.0-json                        7.0.13-0ubuntu0.16.04.1             amd64        JSON module for PHP
ii  php7.0-ldap                        7.0.13-0ubuntu0.16.04.1             amd64        LDAP module for PHP
ii  php7.0-mbstring                    7.0.13-0ubuntu0.16.04.1             amd64        MBSTRING module for PHP
ii  php7.0-mcrypt                      7.0.13-0ubuntu0.16.04.1             amd64        libmcrypt module for PHP
ii  php7.0-opcache                     7.0.13-0ubuntu0.16.04.1             amd64        Zend OpCache module for PHP
ii  php7.0-readline                    7.0.13-0ubuntu0.16.04.1             amd64        readline module for PHP
ii  php7.0-xml                         7.0.13-0ubuntu0.16.04.1             amd64        DOM, SimpleXML, WDDX, XML, and XSL module for PHP
ii  php7.1                             7.1.1-1+deb.sury.org~xenial+1       all          server-side, HTML-embedded scripting language (metapackage)
ii  php7.1-cli                         7.1.1-1+deb.sury.org~xenial+1       amd64        command-line interpreter for the PHP scripting language
ii  php7.1-common                      7.1.1-1+deb.sury.org~xenial+1       amd64        documentation, examples and common module for PHP
ii  php7.1-curl                        7.1.1-1+deb.sury.org~xenial+1       amd64        CURL module for PHP
ii  php7.1-gd                          7.1.1-1+deb.sury.org~xenial+1       amd64        GD module for PHP
ii  php7.1-imap                        7.1.1-1+deb.sury.org~xenial+1       amd64        IMAP module for PHP
ii  php7.1-json                        7.1.1-1+deb.sury.org~xenial+1       amd64        JSON module for PHP
ii  php7.1-opcache                     7.1.1-1+deb.sury.org~xenial+1       amd64        Zend OpCache module for PHP
ii  php7.1-readline                    7.1.1-1+deb.sury.org~xenial+1       amd64        readline module for PHP
ii  php7.1-recode                      7.1.1-1+deb.sury.org~xenial+1       amd64        recode module for PHP
root@DIRECTORY-01:~# dpkg -l | grep smarty
ii  fusiondirectory-smarty3-acl-render 1.0.8.8-3ubuntu2                    all          Provide FusionDirectory ACL based rendering for Smarty3
ii  smarty-gettext                     1.2.0-1ubuntu1                      all          Gettext plugin enabling internationalization in Smarty
ii  smarty3                            3.1.21-1ubuntu1                     all          Template engine for PHP

```

According to several comments on issue #18, the fixes repo's at https://integration.fusiondirectory.org/repos/fixes-releases/debian/ may have fixes for this issue and several other php7 issues however the gpg keys at https://documentation.fusiondirectory.org/en/documentation/admin_get_ggp_keys don't match for the fixes repo. Any idea where I can find those keys?

It was also mentioned in that same issue that having lesser versions than smarty 3.1.29 and smarty-gettext 1.3.0 may cause the issue. There was a debian repo linked for newer versions of those, but I can't find gpg keys for that either.

**Comment by DeMiNe0** [comment]
OK, with the new smarty packages, I seem to be getting template errors during the fusiondirectory setup. I downgraded back to smarty3 3.1.21 and gettext 1.2.0 and that seemed to fix the smarty error. Now I'm getting the following error during setup when trying to connect to my LDAP. I'm going to open a new issue for that though.

**Comment by the-nic** [comment]
In my log I also see the following errors in my syslog:

```
FusionDirectory [unauthenticated]: (view) error: PHP error: Only variables should be passed by reference (/usr/share/fusiondirectory/include/functions.inc, line 589)
FusionDirectory [fd-admin]: (view) error: PHP error: Only variables should be passed by reference (/usr/share/fusiondirectory/include/functions.inc, line 2538)
(view) error: PHP error: Undefined index: SAMBA (/usr/share/fusiondirectory/include/class_filter.inc, line 476)
FusionDirectory []: (view) error: PHP error: Undefined index: tag (/usr/share/fusiondirectory/include/class_filter.inc, line 642)
FusionDirectory [fd-admin]: (view) error: PHP error: Undefined index: MAIL (/usr/share/fusiondirectory/include/class_filter.inc, line 476)
```

and so on.. Does this help?

**Comment by DeMiNe0** [comment]
Okay. I think I managed to switched to smarty3-3.1.29 and smarty-gettex-1.3.0.... It's a dirty work around since I wasn't able to find a built binary for one of them.

```
# Install the regular smarty3 package if you don't have it allready:
apt-get install smarty3
# Install devscripts, needed for dget
apt-get install devscripts
# Make our build dir
mkdir ~/smarty-build/;cd ~/smarty-build;
# Get the source files
dget -x https://launchpad.net/debian/+archive/primary/+files/smarty3_3.1.29-1.dsc
https://launchpad.net/ubuntu/+source/smarty-gettext/1.3.0-1/+build/9855639/+files/smarty-gettext_1.3.0-1_all.deb
wget https://launchpad.net/ubuntu/+source/smarty-gettext/1.3.0-1/+build/9855639/+files/smarty-gettext_1.3.0-1_all.deb
# Install gettext 1.3.0
dpkg -r --force-depends smarty-gettext
dpkg -i smarty-gettext_1.3.0-1_all.deb
# Replace the current smarty3 installation with the .29 installation
cd ~/smarty-build/smarty3-3.1.29/
rm -rf /usr/share/php/smarty3/*
cp -a * /usr/share/php/smarty3/
```

Going to play around in Fusion Directory now and see if that error pops up again.

**Comment by bilbo-the-hobbit** [comment]
Hello,

its seems you have two version of php installed side by side thats not a good idea

ii  php7.0-cli                         7.0.13-0ubuntu0.16.04.1             amd64        command-line interpreter for the PHP scripting language
ii  php7.0-common                      7.0.13-0ubuntu0.16.04.1             amd64        documentation, examples and common module for PHP
ii  php7.0-intl                        7.0.13-0ubuntu0.16.04.1             amd64        Internationalisation module for PHP
ii  php7.0-json                        7.0.13-0ubuntu0.16.04.1             amd64        JSON module for PHP
ii  php7.0-ldap                        7.0.13-0ubuntu0.16.04.1             amd64        LDAP module for PHP
ii  php7.0-mbstring                    7.0.13-0ubuntu0.16.04.1             amd64        MBSTRING module for PHP
ii  php7.0-mcrypt                      7.0.13-0ubuntu0.16.04.1             amd64        libmcrypt module for PHP
ii  php7.0-opcache                     7.0.13-0ubuntu0.16.04.1             amd64        Zend OpCache module for PHP
ii  php7.0-readline                    7.0.13-0ubuntu0.16.04.1             amd64        readline module for PHP
ii  php7.0-xml                         7.0.13-0ubuntu0.16.04.1             amd64        DOM, SimpleXML, WDDX, XML, and XSL module for PHP
ii  php7.1                             7.1.1-1+deb.sury.org~xenial+1       all          server-side, HTML-embedded scripting language (metapackage)
ii  php7.1-cli                         7.1.1-1+deb.sury.org~xenial+1       amd64        command-line interpreter for the PHP scripting language
ii  php7.1-common                      7.1.1-1+deb.sury.org~xenial+1       amd64        documentation, examples and common module for PHP
ii  php7.1-curl                        7.1.1-1+deb.sury.org~xenial+1       amd64        CURL module for PHP
ii  php7.1-gd                          7.1.1-1+deb.sury.org~xenial+1       amd64        GD module for PHP
ii  php7.1-imap                        7.1.1-1+deb.sury.org~xenial+1       amd64        IMAP module for PHP
ii  php7.1-json                        7.1.1-1+deb.sury.org~xenial+1       amd64        JSON module for PHP
ii  php7.1-opcache                     7.1.1-1+deb.sury.org~xenial+1       amd64        Zend OpCache module for PHP
ii  php7.1-readline                    7.1.1-1+deb.sury.org~xenial+1       amd64        readline module for PHP
ii  php7.1-recode                      7.1.1-1+deb.sury.org~xenial+1       amd64        recod

also is the 7.1.1-1+deb.sury.org~xenial+1 official ubuntu packages ?

Cheers

**Comment by thomas955** [comment]
Can confirm this.

**Comment by DeMiNe0** [comment]
Hi,

I noticed that as well from when I was playing around with PHP trying to get things working. I removed all of 7.1:

root@DIRECTORY-01:~# dpkg -l | grep php
ii  libapache2-mod-php                 1:7.0+35ubuntu6                     all          server-side, HTML-embedded scripting language (Apache 2 module) (default)
ii  libapache2-mod-php7.0              7.0.13-0ubuntu0.16.04.1             amd64        server-side, HTML-embedded scripting language (Apache 2 module)
ii  php                                1:7.0+35ubuntu6                     all          server-side, HTML-embedded scripting language (default)
ii  php-cas                            1.3.3-2ubuntu1                      all          Central Authentication Service client library in php
ii  php-cli                            1:7.0+35ubuntu6                     all          command-line interpreter for the PHP scripting language (default)
ii  php-common                         1:35ubuntu6                         all          Common files for PHP packages
ii  php-curl                           1:7.0+35ubuntu6                     all          CURL module for PHP [default]
ii  php-fpdf                           3:1.7.dfsg-1.1ubuntu1               all          PHP class to generate PDF files
ii  php-gd                             1:7.0+35ubuntu6                     all          GD module for PHP [default]
ii  php-imagick                        3.4.0~rc6-1ubuntu3                  amd64        Provides a wrapper to the ImageMagick library
ii  php-imap                           1:7.0+35ubuntu6                     all          IMAP module for PHP [default]
ii  php-intl                           1:7.0+35ubuntu6                     all          Internationalisation module for PHP [default]
ii  php-json                           1:7.0+35ubuntu6                     all          JSON module for PHP [default]
ii  php-ldap                           1:7.0+35ubuntu6                     all          LDAP module for PHP [default]
ii  php-mbstring                       1:7.0+35ubuntu6                     all          MBSTRING module for PHP [default]
ii  php-mcrypt                         1:7.0+35ubuntu6                     all          libmcrypt module for PHP [default]
ii  php-mdb2                           2.5.0b5-1build1                     all          merge of the PEAR DB and Metabase php database abstraction layers
ii  php-pear                           1:1.10.1+submodules+notgz-6         all          PEAR Base System
ii  php-readline                       1:7.0+35ubuntu6                     all          readline module for PHP [default]
ii  php-recode                         1:7.0+35ubuntu6                     all          recode module for PHP [default]
ii  php-xml                            1:7.0+35ubuntu6                     all          DOM, SimpleXML, WDDX, XML, and XSL module for PHP [default]
ii  php7.0                             7.0.13-0ubuntu0.16.04.1             all          server-side, HTML-embedded scripting language (metapackage)
ii  php7.0-cli                         7.0.13-0ubuntu0.16.04.1             amd64        command-line interpreter for the PHP scripting language
ii  php7.0-common                      7.0.13-0ubuntu0.16.04.1             amd64        documentation, examples and common module for PHP
ii  php7.0-curl                        7.0.13-0ubuntu0.16.04.1             amd64        CURL module for PHP
ii  php7.0-gd                          7.0.13-0ubuntu0.16.04.1             amd64        GD module for PHP
ii  php7.0-imap                        7.0.13-0ubuntu0.16.04.1             amd64        IMAP module for PHP
ii  php7.0-intl                        7.0.13-0ubuntu0.16.04.1             amd64        Internationalisation module for PHP
ii  php7.0-json                        7.0.13-0ubuntu0.16.04.1             amd64        JSON module for PHP
ii  php7.0-ldap                        7.0.13-0ubuntu0.16.04.1             amd64        LDAP module for PHP
ii  php7.0-mbstring                    7.0.13-0ubuntu0.16.04.1             amd64        MBSTRING module for PHP
ii  php7.0-mcrypt                      7.0.13-0ubuntu0.16.04.1             amd64        libmcrypt module for PHP
ii  php7.0-opcache                     7.0.13-0ubuntu0.16.04.1             amd64        Zend OpCache module for PHP
ii  php7.0-readline                    7.0.13-0ubuntu0.16.04.1             amd64        readline module for PHP
ii  php7.0-recode                      7.0.13-0ubuntu0.16.04.1             amd64        recode module for PHP
ii  php7.0-xml                         7.0.13-0ubuntu0.16.04.1             amd64        DOM, SimpleXML, WDDX, XML, and XSL module for PHP

**Comment by DeMiNe0** [comment]
Even now with the single PHP version from the official repo's it's still giving me the Unknown element type specified! error. This happened to me after I removed all the departments. It shows up on the Users page. When I go to the departments page it shows the following error:
`Fatal error: Uncaught NonExistingObjectTypeException: Non-existing type "DEPARTMENT" in /usr/share/fusiondirectory/include/class_objects.inc:205 Stack trace: #0 /usr/share/fusiondirectory/include/class_listing.inc(230): objects::infos('DEPARTMENT') #1 /usr/share/fusiondirectory/include/class_listing.inc(81): listing->load(Array) #2 /usr/share/fusiondirectory/include/simpleplugin/class_simpleManagement.inc(188): listing->__construct(Array) #3 /usr/share/fusiondirectory/plugins/admin/departments/class_departmentManagement.inc(50): simpleManagement->__construct() #4 /usr/share/fusiondirectory/plugins/admin/departments/main.inc(43): departmentManagement->__construct() #5 /usr/share/fusiondirectory/html/main.php(286): require('/usr/share/fusi...') #6 {main} thrown in /usr/share/fusiondirectory/include/class_objects.inc on line 205`

**Comment by thomas955** [comment]
A temporary solutions seems to remove the /var/lib/php/session/* .

But these errors still occure!

**Comment by MCMic** [comment]
Yeah the problem seems to be that the filter for systems (or sometimes an other plugin) gets mixed up with the one for users in the cache of the session.
Looking at the code I can’t figure out why and I never managed to reproduce it yet.

It seems specific to PHP 7 and our PHP 7 test suite is waiting for Debian to fix https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=847571

**Comment by bilbo-the-hobbit** [comment]
hello,

yes you need to use the fixes version that will become the next stable to get rid of most of errors

https://documentation.fusiondirectory.org/en/documentation_testing#testing-with-the-fixes-versions

Install this one then report back to us

Cheers

**Comment by DeMiNe0** [comment]
I've been using the fixes repo at https://integration.fusiondirectory.org/repos/fixes-releases/debian/fusiondirectory-119-fixes-jessie/ for the past week or so. All of my previous error reports have been on that repo.

**Comment by DeMiNe0** [comment]
I've since completely re-rolled my FusionDirectory installation on Debian Jessie due to deadline constraints in my company. Everything works fine on Debian Jessie running Jessie's default php version of 5.6. Been working with it for about an hour trying relentlessly to get that error to show back up. I can't.

I think in order to successfully reproduce this issue you'll need to install FusionDirectory on top of Ubuntu 16.04. I believe that's where this issue is happening in a highly reproducible state.

**Comment by MCMic** [comment]
Ok so it seemed to be linked to the use of references to session vars in some parts of the code, so I removed as much of them as possible and the problem seems to be gone in the development version: https://integration.fusiondirectory.org/repos/development/debian/fusiondirectory-dev-jessie/
So it should be fixed in 1.1 when it will be out. (Note that the development version should work if you want to use it but does not support samba winstations as of today. Also if you use systems plugins you will need to migrate entries with fusiondirectory-setup)

**Comment by bilbo-the-hobbit** [comment]
hello,

lots of fixes have been put in the 1.0.20 release that will go out this week.

Closing this
